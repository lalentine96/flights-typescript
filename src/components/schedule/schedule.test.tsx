import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Schedule from './schedule';

afterEach(cleanup);

it('should match the snapshot', () => {
    const flights = [
        {
            scheduledTime: "2020-04-10 19:09+03:00",
            actualTime: "2020-04-10 18:56+03:00",
            city: "DÃ¼sseldorf",
            airline: "Emirates Airlines",
            number: "AV 3146",
            terminal: "A",
            status: "Arrived",
        },
        {
            scheduledTime: "2020-04-10 19:20+03:00",
            actualTime: "2020-04-10 19:01+03:00",
            city: "Paris",
            airline: "UTAIR",
            number: "N4 8795",
            terminal: "C",
            status: "Delayed"
        }
    ]

    const { asFragment } = render(<Schedule flights={flights}/>);
    expect(asFragment()).toMatchSnapshot();
});
