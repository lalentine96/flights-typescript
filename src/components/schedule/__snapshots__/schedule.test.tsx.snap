// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <table
    class="table table-striped"
  >
    <thead
      class="thead-light"
    >
      <tr>
        <th>
          Scheduled time
        </th>
        <th>
          Actual time
        </th>
        <th>
          City
        </th>
        <th>
          Airline
        </th>
        <th>
          Flight number
        </th>
        <th>
          Terminal
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          19:09
        </td>
        <td>
          18:56
        </td>
        <td>
          DÃ¼sseldorf
        </td>
        <td>
          Emirates Airlines
        </td>
        <td>
          AV 3146
        </td>
        <td>
          A
        </td>
        <td>
          Arrived
        </td>
      </tr>
      <tr>
        <td>
          19:20
        </td>
        <td>
          19:01
        </td>
        <td>
          Paris
        </td>
        <td>
          UTAIR
        </td>
        <td>
          N4 8795
        </td>
        <td>
          C
        </td>
        <td>
          Delayed
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
